cmd/antrea-agent/options.go

```diff
@@ -273,3 +277,21 @@ func (o *Options) validateFlowExporterConfig() error {
 	}
 	return nil
 }
+
+func (o *Options) validateFlexibleIPAMConfig() error {
+	if features.DefaultFeatureGate.Enabled(features.FlexibleIPAM) {
+		if o.config.OVSDatapathType != string(ovsconfig.OVSDatapathSystem) {
+			return fmt.Errorf("FlexibleIPAM requires 'system' OVSDatapathType")
+		}
+		if o.config.TrafficEncapMode != config.TrafficEncapModeNoEncap.String() {
+			return fmt.Errorf("FlexibleIPAM requires 'NoEncap' TrafficEncapMode")
+		}
+		if !o.config.NoSNAT {
+			return fmt.Errorf("FlexibleIPAM requires NoSNAT")
+		}
+		if runtime.IsWindowsPlatform() {
```

Comment:
Isn't this check covered with antrea_features?

Reply:
Correct, we can remove this. Thanks!